### ISSUE TITLE
Compiling without taint support emits new build-time warning

### ISSUE THREAD

**Issue by jkeenan** [Perl/perl5#19654]
When I built `perl` at recent commit 99db5f9692 with the new option to *not* have taint support, one previously unobserved build-time warning was generated during `make test_prep`.  I configured like this:
```
$ sh ./Configure -des -Dusedevel -Uversion_only -Utaint_support
```
... and got this build-time warning:
```
$ report-build-warnings 99db5f9692.freebsd.threaded.notaint.maketp.output.txt.gz
File:  99db5f9692.freebsd.threaded.notaint.maketp.output.txt.gz

  Wunused-variable                           1

$ parse-build-warnings 99db5f9692.freebsd.threaded.notaint.maketp.output.txt.gz
File:  99db5f9692.freebsd.threaded.notaint.maketp.output.txt.gz

[
  {
    char   => 15,
    group  => "Wunused-variable",
    line   => 9393,
    source => "regcomp.c",
    text   => "unused variable 'oldtainted'",
  },
]
```
Relevant part of `make test_prep` output:
```
cc -c -DPERL_CORE -DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -DSILENT_NO_TAINT_SUPPORT -std=c99 -O2 -Wall -Werror=pointer-arith -Werror=vla -Wextra -Wno-long-long -Wno-declaration-after-statement -Wc++-compat -Wwrite-strings regcomp.c
regcomp.c:9393:15: warning: unused variable 'oldtainted' [-Wunused-variable]
    const int oldtainted = TAINT_get;
              ^
1 warning generated.
```
The statement in `regcomp.c` was added in this commit:
```
commit accb4364d92e26c20e6a538fc04d1af52a8b94e2
Author:     Karl Williamson <khw@cpan.org>
AuthorDate: Tue Feb 23 14:04:19 2016 -0700
Commit:     Karl Williamson <khw@cpan.org>
CommitDate: Tue Feb 23 23:59:31 2016 -0700
```
However, the warning is **not** being emitted on regular (with taint support) builds, so we wouldn't have been able to observe it until we more recently added the option to build without taint support (`-Utaint_support`).

@khwilliamson, can you take a look?

**Note:** Since we're now in code freeze (I hope!) and since compiling with no build-time warnings is a "nice to have" rather than a strict requirement for a production release -- though it is a "**very** nice to have" for a **production** release -- I'll leave it up to the production release manager to determine whether we can accept a p.r. correcting this problem.  @rjbs

Thank you very much.
Jim Keenan

**Comment by demerphq** [comment]
This is trivial to fix. I'll post a patch today assuming @khwilliamson doesnt push one before I do.

**Comment by demerphq** [comment]
This is fixed by PR #19656

**Comment by khwilliamson** [comment]
Fixed by https://github.com/Perl/perl5/commit/70f5c3e99f92075109e4262faf825fb00ea79bd3

**Comment by jkeenan** [comment]
> Fixed by [70f5c3e](https://github.com/Perl/perl5/commit/70f5c3e99f92075109e4262faf825fb00ea79bd3)

Confirmed.
```
$ ./perl -Ilib -MConfig -E 'say "taint: ", !exists($Config{taint_support}) || $Config{taint_support} ? "yes" : "no";'
taint: no
$ report-build-warnings 70f5c3e99f.freebsd.threaded.notaint.maketp.output.txt.gz 
File:  70f5c3e99f.freebsd.threaded.notaint.maketp.output.txt.gz
```
